extensions/positron-r/src/r-installation.ts

```diff
@@ -25,20 +25,82 @@ export interface RMetadataExtra {
 	 * https://github.com/posit-dev/positron/issues/2659
 	 */
 	readonly current: boolean;
+
+	/**
+	 * How did we discover this R binary?
+	 */
+	readonly reasonDiscovered: ReasonDiscovered[] | null;
+}
+
+/**
+ * Enum represents how we discovered an R binary.
+ */
+export enum ReasonDiscovered {
+	affiliated = "affiliated",
+	registry = "registry",
+	/* eslint-disable @typescript-eslint/naming-convention */
+	PATH = "PATH",
+	HQ = "HQ",
+	/* eslint-enable @typescript-eslint/naming-convention */
+	adHoc = "adHoc",
+	user = "user"
+}
+
+/**
+ * Enum represents why we rejected an R binary.
+ */
+export enum ReasonRejected {
+	invalid = "invalid",
+	unsupported = "unsupported",
+	nonOrthogonal = "nonOrthogonal"
+}
```

Comment:
These really do need to be string enums, so that, when `reason` arrives in `friendlyReason()` below, we can use its value to determine which enum we're dealing with. This seems to be a fundamental awkwardness with enums in javascript (?), https://stackoverflow.com/q/63753808.